src/types/auth.ts

```diff
@@ -1,8 +1,20 @@
 import {AvataaarProps} from "components/Avatar";
 
+export const ACCOUNT_TYPE_ANONYMOUS = "ANONYMOUS";
+
+// as retrieved from backend
+export interface AuthDto {
+  id: string;
+  name: string;
+  accountType: string;
+  avatar?: AvataaarProps;
+}
+
+// as used in frontend
 export interface Auth {
   id: string;
   name: string;
+  isAnonymous: boolean;
   avatar?: AvataaarProps;
 }
```

Comment:
Imo we could add the accountType to Auth instead of isAnonymouse and remove the AuthDto. To check if an account is anonymous we could perform this check: `accountType != ACCOUNT_TYPE_ANONYMOUS`. But if you think it's better to handle it this way, it works for me too 😄.

Reply:
Where exactly would you perform that check? I think you need the second type because it gets sent like that from the backend; so either perform the cast like I did it the frontend or you change the backend to send `isAnonymous` instead of `accountType`. Just changing the property name in `Auth` would likely result in a type error somewhere.

Reply:
I do agree though it looks kinda odd so maybe a backend change is the way to go

Reply:
Everywhere where you need the isAnonymous variable you could perform the `accountType != ACCOUNT_TYPE_ANONYMOUS` check.
But I guess it's not the cleanest solution either.
I don't think changing the backend is a better solution. Just like in the frontend you  would have another user struct wich has the isAnonymous variable instead of AccountType.

Reply:
Yeah, that's reason I opted for using another variable in the first place, because the frontend just needs to differentiate between anonymous and non-anonymous users. Whether the authentification used is Google, Apple or whatever should be irrelevant and handling a boolean is easier than a string 😄   
That being said I'll leave it like this for now, changing it in the future is easy enough.